core/src/com/unciv/logic/civilization/CivilizationInfo.kt

```diff
@@ -715,6 +715,18 @@ class CivilizationInfo : IsPartOfGameInfoSerialization {
             }
     }
 
+    fun doMoveAutomatedUnits() {
+        for (unit in units)
+            unit.doAction()
+        hasMovedAutomatedUnits = true
+    }
+
+    fun canMoveAutomatedUnits(): Boolean {
+        return !hasMovedAutomatedUnits && units.any {
+            it.currentMovement > Constants.minimumMovementEpsilon
```

Comment:
```suggestion
            it.currentMovement > 0f
```

Should never be epsilon since we deal with that in the unit movement